### PR TITLE
fix: cl merge main swaprouter changes (state-breaking)

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

These changes are required to wire up Roman's PR of merging concentrated-liquidity-main branch to main. This PR is separated since it touches logic other than strictly the x/concentrated liquidity folder, and we should therefore scrutinize this PR more than the original merge to main PR.

## Brief Changelog

- Adds cl to app keepers
- Adds cl entry into swaprouter keeper
- Adds cl as an app module
- Adds cl to the v14 store upgrade handler (we may punt this to v15 but it exists there now due to e2e)
- Removes events emitted from the cl msg server
- Removes redundnat validation in x/swaprouter's `CreatePool`. Explanations for each added:
   * https://github.com/osmosis-labs/osmosis/pull/3876#discussion_r1058490778
   * https://github.com/osmosis-labs/osmosis/pull/3876#discussion_r1058496005

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -36,6 +36,8 @@ cloud.google.com/go/storage v1.6.0/go.mod h1:N7U0C8pVQ/+NIKOBQyamJIeKQKkZ+mxpohl
 cloud.google.com/go/storage v1.8.0/go.mod h1:Wv1Oy7z6Yz3DshWRJFhqM/UCfaWIRTdp0RXyy7KQOVs=
 cloud.google.com/go/storage v1.10.0/go.mod h1:FLPqc6j+Ki4BU591ie1oL6qBQGu2Bl/tZ9ullr3+Kg0=
 cloud.google.com/go/storage v1.14.0/go.mod h1:GrKmX003DSIwi9o29oFT7YDnHYwZoctc3fOKtUw0Xmo=
+cosmossdk.io/errors v1.0.0-beta.7 h1:gypHW76pTQGVnHKo6QBkb4yFOJjC+sUGRc5Al3Odj1w=
+cosmossdk.io/errors v1.0.0-beta.7/go.mod h1:mz6FQMJRku4bY7aqS/Gwfcmr/ue91roMEKAmDUDpBfE=
```
Comment (p0mvn) [inline go.sum:40]: Why do we need these?

**Comment by p0mvn** [comment]
Please do not merge until: https://github.com/osmosis-labs/osmosis/pull/3868 is merged

**Review by p0mvn — APPROVED** [review]
LGTM once https://github.com/osmosis-labs/osmosis/pull/3868 is merged

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM! We should try to get this merged asap now that #3868 is merged so we don't diverge too much on other CL PRs to main

**Pull request merged**